### PR TITLE
Restore warning commit status

### PR DESCRIPTION
**Pull request by silverwind**
Partial revert of https://github.com/go-gitea/gitea/pull/25839. This commit status is used by a number of external integrations, so I think we should not remove it (See https://github.com/go-gitea/gitea/pull/25839#issuecomment-1729002077). This is a rare case where an existing migration needed to be alterted to avoid data loss.

### CODE REVIEW

File: modules/structs/commit_status.go [hunk 1]
```diff
@@ -16,13 +16,16 @@ const (
...
+	// CommitStatusWarning is for when the CommitStatus is Warning
+	CommitStatusWarning CommitStatusState = "warning"
 )
 
 var commitStatusPriorities = map[CommitStatusState]int{
 	CommitStatusError:   0,
 	CommitStatusFailure: 1,
 	CommitStatusPending: 2,
 	CommitStatusSuccess: 3,
+	CommitStatusWarning: 4,
```
Comment (silverwind) [inline modules/structs/commit_status.go]: Not 100% sure about this, see previous code for reference:

https://github.com/go-gitea/gitea/pull/25839/files#diff-28c2d52083459d45d94ce384950fe4a3bddacdf125a296872a0b3f68734a1b16L27-R27

Comment (lunny) [reply modules/structs/commit_status.go]: Maybe 

CommitStatusWarning: 3,
CommitStatusSuccess: 4,

**Comment by yardenshoham** [comment]
Don't forget https://github.com/go-gitea/gitea/blob/e3229c8e9bd506fce528ee77b74b5a64f66e8e6e/web_src/js/components/DashboardRepoList.vue#L10-L15

**Comment by silverwind** [comment]
> Don't forget
> 
> https://github.com/go-gitea/gitea/blob/e3229c8e9bd506fce528ee77b74b5a64f66e8e6e/web_src/js/components/DashboardRepoList.vue#L10-L15

Done. Interestingly https://github.com/go-gitea/gitea/pull/25839 did not touch this file and blame info is also inconclusive. Maybe it was missing a long time already in the JS.

**Comment by yardenshoham** [comment]
It's on https://github.com/go-gitea/gitea/pull/26036

**Comment by silverwind** [comment]
Thanks. I also reverted https://github.com/go-gitea/gitea/pull/26120 which was a followup to https://github.com/go-gitea/gitea/pull/25839.

File: modules/structs/commit_status.go (hunk 1 above)
Comment (delvh) [inline modules/structs/commit_status.go]: I think it is supposed to be
```suggestion
	CommitStatusWarning: 2,
	CommitStatusPending: 3,
	CommitStatusSuccess: 4,
```
(assuming we want to say that a `warning` state is worse than a still pending status as the pending status might also be successful)

Comment (delvh) [reply modules/structs/commit_status.go]: Yep, I just double-checked, that's also the previous behavior.

Comment (silverwind) [reply modules/structs/commit_status.go]: Thanks, I was getting very confused at that old code. Let's see if tests pass with this change.

Comment (silverwind) [reply modules/structs/commit_status.go]: Likely fixed with https://github.com/go-gitea/gitea/pull/27504/commits/89269758fe70a5f9d6986c8bed659c1d9fd8a246.

Comment (lunny) [reply modules/structs/commit_status.go]: The greater among statuses looks still wired.

**Review by delvh — APPROVED** [review]

File: modules/structs/commit_status.go (hunk 1 above)
Comment (silverwind) [reply modules/structs/commit_status.go]: @delvh any opinion?

Comment (delvh) [reply modules/structs/commit_status.go]: I've simply compared it with the previous code, and I think a `warning` state is worse than a `pending` state as the `warning` state is guaranteed to be worse than a potentially successful `pending` state

**Review by denyskon — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

File: modules/structs/commit_status.go (hunk 1 above)
Comment (silverwind) [reply modules/structs/commit_status.go]: If I interpret the previous correctly, it should be ranked:

CommitStatusError
CommitStatusFailure
CommitStatusWarning
CommitStatusPending
CommitStatusSuccess

Comment (delvh) [reply modules/structs/commit_status.go]: Yes, that's exactly what I said

Comment (silverwind) [reply modules/structs/commit_status.go]: I see the branch was already correct. Let's merge it.

**Pull request merged**